hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -604,6 +606,27 @@ evalEither = \case
   Right x ->
     pure x
 
+-- | Fails the test if the 'Either' is 'Right', otherwise returns the value in
+--   the 'Left'.
+--
+evalLeft :: (MonadTest m, Show a, HasCallStack) => Either x a -> m x
+evalLeft = \case
+  Left x ->
+    pure x
+  Right x ->
+    withFrozenCallStack $ failWith Nothing $ showPretty x
+
+-- | Fails the test if the 'Either' is 'Right'.
+-- An example use case would be testing that a parser the returns an 'Either'
+-- fails on all invalid inputs.
+--
+isLeft :: (MonadTest m, Show a, HasCallStack) => Either x a -> m ()
+isLeft = \case
+  Left _ ->
+    pure ()
+  Right x ->
+    withFrozenCallStack $ failWith Nothing $ showPretty x
```

Comment:
Wondering if this duplication tries to tell us something :confused: 

![image](https://user-images.githubusercontent.com/287532/40573843-5d90013c-60d0-11e8-91a9-5f19d90ea870.png)

Reply:
Updated the PR with more succinct versions.